src/bigint.rs

```diff
@@ -879,6 +883,65 @@ impl BigUint {
 
     /// Creates and initializes a `BigUint`.
     ///
+    /// The bytes are in big-endian byte order
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use num::bigint::BigUint;
+    ///
+    /// assert_eq!(BigUint::from_bytes_be("A".as_bytes()),
+    ///            BigUint::parse_bytes("65".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("AA".as_bytes()),
+    ///            BigUint::parse_bytes("16705".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("AB".as_bytes()),
+    ///            BigUint::parse_bytes("16706".as_bytes(), 10).unwrap());
+    /// assert_eq!(BigUint::from_bytes_be("Hello world!".as_bytes()),
+    ///            BigUint::parse_bytes("22405534230753963835153736737".as_bytes(), 10).unwrap());
+    /// ```
+    #[inline]
+    pub fn from_bytes_be(msg: &[u8]) -> BigUint {
+        BigUint::parse_bytes(msg.to_hex().as_bytes(), 16).unwrap()
+    }
+
+    /// Creates and initializes a `BigUint`.
+    ///
+    /// The bytes are in little-endian byte order
+    #[inline]
+    pub fn from_bytes_le(msg: &[u8]) -> BigUint {
+        let mut s = msg.to_hex();
+        unsafe {
+            let v = s.as_mut_vec();
+            v.reverse();
+        }
+        BigUint::parse_bytes(s.as_bytes(), 16).unwrap()
+    }
+
+    /// Returns the byte representation of the `BigUint` in little-endian byte order.
+    #[inline]
+    pub fn to_bytes_le(&self) -> Vec<u8> {
+        let mut result = Vec::new();
+        for word in self.data.iter() {
+            let mut w = *word;
+            while w != 0 {
```

Comment:
(Another special case: should `0.to_bytes_le()` give `[0]` or `[]`? I'm inclined to say the former, even though that isn't the shortest representation that `from_bytes_le` will parse.)